src/transaction/transaction.cpp

```diff
@@ -75,15 +92,30 @@ Transaction::Transaction() :
 void Transaction::sign(Signer const &signer)
 {
     if (m_signature != nullptr)
+    {
         m_signature = nullptr;
+    }
 
-    std::string const txSerialized = Transaction::serialize();
+    std::string const txSerialized = internal::getTxSerializedHashIfRequired(*this);
     std::string const tmpSign = signer.getSignature(txSerialized);
     std::string const signature = util::stringToHex(tmpSign);
 
     m_signature.reset(new std::string(signature));
 }
 
+bool Transaction::verify()
+{
+    if (m_signature == nullptr) throw std::runtime_error(ERROR_MSG_SIGNATURE);
+    if (m_sender == nullptr) throw std::runtime_error(ERROR_MSG_SENDER);
+
+    auto tmp = m_signature;
+    m_signature = nullptr;
```

Comment:
Oh :)

This is because you want to alter the behavior of `serialize()`, right? That fact is not extremely obvious right now, do you think you can perhaps add a comment here to describe the workaround?

Reply:
> This is because you want to alter the behavior of serialize(), right?

Yes, because I want to verify the **signature** of the **message**(the serialized transaction, **without signature**).
Will add a comment here